### DISCUSSION TITLE
Linker errors after copying Imgui files to my codebase

### DISCUSSION THREAD

**Discussion by Tonez3D** [ocornut/imgui#5983] [New Users Build/Link/Run issues ONLY!]
I want to use ImGui in a program of mine. It's not my first time working with guis but it's the first time I've worked with an api like this one. 
I brought the imgui files I thought necessary into the visual studio folder for my project:
![issue1file](https://user-images.githubusercontent.com/111995183/206890531-a9851182-9c76-4415-8b6d-cbec4001afef.PNG)

Im using version 1.89 with directX12
Visual Studio 2019 on Windows 10
With C++14
The errors:
![issue2err](https://user-images.githubusercontent.com/111995183/206890579-b0947ee7-6acb-4f52-b9ad-41e34d7ad658.PNG)

I don't know much about compiling since I'm used to using python the little I do know comes from java. 

If someone could help with the linker issues that would be nice. If possible a little explanation on what caused it or how the linker works is also welcome too.

**Comment by ocornut** [comment]
Linkers errors are the same everywhere on the internet: you are not compiling and linking with the source files.  You need to add them to your visual studio project, not only to a folder.

**Comment by Tonez3D** [comment]
Thanks for the reply. I added all these imgui files to the project:
![new](https://user-images.githubusercontent.com/111995183/206927876-7136d675-a149-4b47-8d0f-7bdee3fe429f.PNG)

The number of errors went down but these are still there:
![red](https://user-images.githubusercontent.com/111995183/206927944-de0d1a0e-7a7b-4478-9847-b77666447fde.PNG)
Are there Imgui files a forgot add to project or am I adding them incorrectly?

**Comment by ocornut** [comment]
Look what the missing symbols are and try to guess which library they are from, or google them.

**Comment by Tonez3D** [comment]
I googled the function names and how to link libraries and was able to find out what directx12 files I needed to link. Thank you for the help. Would seeing what I changed imgui_impl_dx12.cpp help you develop ImGui in any way?

**Reply by PathogenDavid** [reply]
You should not need to modify `imgui_impl_dx12.cpp`. You should add the D3D12 import libraries (IE: `d3d12.lib` and `dxgi.lib`) to your project configuration.

**Reply by Tonez3D** [reply]
With `#pragma comment(lib, "library name")` or would it be done in a different way?

**Reply by ocornut** [reply]
There are multiple answers. Search “how to link a library with visual studio”.

**Reply by Tonez3D** [reply]
Thanks I got it working.
